docs/madr/decisions/031-locality-aware-load-balancing.md

```diff
@@ -0,0 +1,300 @@
+        fallback: Any # (5)
+```
+
+
+(1) In `inZone` section, you configure the priorities of tags used for load balancing. If list is empty or inZone is not specified, load balancing works normally, without taking any priorities into account. 
+
+(2) In `crossZone` section, you configure zones priority.
+
+(3) In `rules` section, you configure list of zone groups priorities, first groups will have the highest load balancing priority and last in order will have the lowest priority.
+
+(4) In `groups` section, you configure logical zone groups. Let's assume `zone-1` and `zone-2` are located in the same datacenter, so you want to load balance equally between them. `Zone-5` is located in another datacenter so it should have lower priority, that is why it is in the second groups entry.
+
+(5) At last we should apply some fallback if no dataplanes from previous configuration are available, at this moment we allow `Fail` and `Any` fallback. Default value `Any`'
+
+### API examples based on use cases
+
+#### Keeping traffic in same zone (disable cross zone traffic)
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+```
+
+![Use case 1](assets/031/use_case_1.png)
+
+#### Load balancing based on priorities within a zone. Example: node → AZ → region
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: ["k8s.io/node", "k8s.io/az"]
+```
+
+![Use case 2](assets/031/use_case_2.png)
+![Use case 2](assets/031/use_case_2.png)
+![Use case 2](assets/031/use_case_2.png)
+
+#### Prefer traffic in same zone with fallback to any other zone
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+      crossZone:
+        fallback: Any
+```
+
+![Use case 3](assets/031/use_case_3_1.png)
+![Use case 3](assets/031/use_case_3_2.png)
+
+#### Load balance equally to list of zones
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+      crossZone:
+        - groups:
+          - zones: ["zone-1", "zone-3"]
+        fallback: Fail
+```
+
+![Use case 4](assets/031/use_case_4_1.png)
+![Use case 4](assets/031/use_case_4_2.png)
+![Use case 4](assets/031/use_case_4_3.png)
+
+#### When leaving zone, target specific zones first
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+      crossZone:
+        - groups:
+          - zones: ["zone-1"]
+        - groups:
+          - zones: ["zone-3"]
+```
+
+![Use case 5](assets/031/use_case_5_1.png)
+![Use case 5](assets/031/use_case_5_2.png)
+![Use case 5](assets/031/use_case_5_3.png)
+
+#### Exclude single zone from load balancing
```

Comment:
The description is not really true. If I add `zone-4` it will also be excluded. I think we don't really have a way to exclude just a set of zone atm.

Reply:
Actually just saw there's further comments lower down